HLTrigger/special/plugins/HLTCTPPSLocalTrackFilter.cc

```diff
@@ -0,0 +1,234 @@
+{
+  edm::ParameterSetDescription desc;
+  makeHLTFilterDescription(desc);
+
+  desc.add<edm::InputTag>("pixelLocalTrackInputTag",   edm::InputTag("ctppsPixelLocalTracks"))
+    ->setComment("input tag of the pixel local track collection");
+  desc.add<edm::InputTag>("stripLocalTrackInputTag",   edm::InputTag("totemRPLocalTrackFitter"))
+    ->setComment("input tag of the strip local track collection");
+  desc.add<edm::InputTag>("diamondLocalTrackInputTag", edm::InputTag("ctppsDiamondLocalTracks"))
+    ->setComment("input tag of the diamond local track collection");
+
+  desc.add<unsigned int>("detectorBitset", static_cast<unsigned int>(1))
+    ->setComment("bitset of which detector types to consider: bit 1 -> pixel; bit 2 -> strips; bit 3 -> diamonds. eg. the value 1 will only consider the pixel detectors, the value 5 will consider the diamond and pixel detectors");
+
+  desc.add<int>("minTracks", 2)
+    ->setComment("minimum number of tracks");
+  desc.add<int>("minTracksPerArm", 1)
+    ->setComment("minimum number of tracks per arm of the CTPPS detector");
+
+  desc.add<int>("maxTracks", -1)
+    ->setComment("maximum number of tracks, if smaller than minTracks it will be ignored");
+  desc.add<int>("maxTracksPerArm", -1)
+    ->setComment("maximum number of tracks per arm of the CTPPS detector, if smaller than minTrackPerArm it will be ignored");
+  desc.add<int>("maxTracksPerPot", -1)
+    ->setComment("maximum number of tracks per roman pot of the CTPPS detector, if negative it will be ignored");
+
+  desc.add<int>("triggerType", trigger::TriggerTrack);
+
+  descriptions.add("hltCTPPSLocalTrackFilter", desc);
+  return;
+}
+
+//
+// destructor and constructor
+//
+HLTCTPPSLocalTrackFilter::~HLTCTPPSLocalTrackFilter()= default;
+
+HLTCTPPSLocalTrackFilter::HLTCTPPSLocalTrackFilter(const edm::ParameterSet& iConfig):
+  HLTFilter(iConfig),
+  pixelLocalTrackInputTag_   (iConfig.getParameter< edm::InputTag > ("pixelLocalTrackInputTag")),
+  stripLocalTrackInputTag_   (iConfig.getParameter< edm::InputTag > ("stripLocalTrackInputTag")),
+  diamondLocalTrackInputTag_ (iConfig.getParameter< edm::InputTag > ("diamondLocalTrackInputTag")),
+  detectorBitset_            (iConfig.getParameter< unsigned int  > ("detectorBitset")),
+  minTracks_                 (iConfig.getParameter< int           > ("minTracks")),
+  minTracksPerArm_           (iConfig.getParameter< int           > ("minTracksPerArm")),
+  maxTracks_                 (iConfig.getParameter< int           > ("maxTracks")),
+  maxTracksPerArm_           (iConfig.getParameter< int           > ("maxTracksPerArm")),
+  maxTracksPerPot_           (iConfig.getParameter< int           > ("maxTracksPerPot")),
+  usePixel_   (false),
+  useStrip_   (false),
+  useDiamond_ (false)
+{
+  if(detectorBitset_ & 1)
+    usePixel_ = true;
+  if(detectorBitset_ & 2)
+    useStrip_ = true;
+  if(detectorBitset_ & 4)
+    useDiamond_ = true;
+
```

Comment:
The above is horrible, simply use these three Booleans as bool module parameters and remove detectorBitset.